src/test/java/org/cactoos/experimental/ThreadsTest.java

```diff
@@ -25,65 +25,103 @@
-import org.cactoos.scalar.ScalarOf;
+import org.junit.jupiter.api.RepeatedTest;
 import org.junit.jupiter.api.Test;
 import org.llorllale.cactoos.matchers.Assertion;
 import org.llorllale.cactoos.matchers.HasValues;
 import org.llorllale.cactoos.matchers.Throws;
+import org.llorllale.cactoos.shaded.org.cactoos.scalar.LengthOf;
 
 /**
  * Test case for {@link Threads}.
  *
  * @since 1.0.0
- * @checkstyle MagicNumberCheck (500 lines)
  * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
  */
 @SuppressWarnings("PMD.AvoidDuplicateLiterals")
 final class ThreadsTest {
 
+    /**
+     * Repetitions when running test several times.
+     */
+    private static final int REPETITIONS = 5;
+
     /**
      * Execute the tasks concurrently using {@link Threads} when
      *  {@link ExecutorService} was initiated by someone else.
      */
-    @Test
-    void containsResults() {
-        this.repeat(
-            arg -> {
-                final ExecutorService extor = Executors.newFixedThreadPool(3);
-                try {
-                    new Assertion<>(
-                        "contains results from callables",
-                        new Threads<String>(
-                            extor,
-                            () -> {
-                                this.sleep();
-                                return "txt 1";
-                            },
-                            () -> {
-                                this.sleep();
-                                return "txt 2";
-                            },
-                            () -> {
-                                this.sleep();
-                                return "txt 3";
-                            }
-                        ),
-                        new HasValues<>("txt 1", "txt 2", "txt 3")
-                    ).affirm();
-                } finally {
-                    extor.shutdown();
-                    if (!extor.awaitTermination(1L, TimeUnit.SECONDS)) {
-                        extor.shutdownNow();
+    @RepeatedTest(ThreadsTest.REPETITIONS)
+    void containsResults() throws Exception {
+        final ExecutorService extor = Executors.newFixedThreadPool(3);
+        try {
+            new Assertion<>(
+                "Must contain results from callables",
+                new Threads<String>(
+                    extor,
+                    Duration.ofSeconds(1),
+                    () -> {
+                        this.sleep();
+                        return "txt 1";
+                    },
+                    () -> {
+                        this.sleep();
+                        return "txt 2";
+                    },
+                    () -> {
+                        this.sleep();
+                        return "txt 3";
                     }
-                }
-            }
-        );
+                ),
+                new HasValues<>(
+                    "txt 1",
```

Comment:
@victornoel we could avoid to repeat these three values each time (txt 1, txt 2 and txt 3) by declaring an iterable variable of these values.